src/librustc_typeck/check/method/suggest.rs

```diff
@@ -752,20 +752,21 @@ fn compute_all_traits<'a, 'gcx, 'tcx>(tcx: TyCtxt<'a, 'gcx, 'tcx>) -> Vec<DefId>
                            traits: &mut Vec<DefId>,
                            external_mods: &mut FxHashSet<DefId>,
                            def: Def) {
-        let def_id = def.def_id();
-        match def {
-            Def::Trait(..) => {
-                traits.push(def_id);
-            }
-            Def::Mod(..) => {
-                if !external_mods.insert(def_id) {
-                    return;
+        if let Some(def_id) = def.opt_def_id() {
+            match def {
+                Def::Trait(..) => {
```

Comment:
Instead of calling the `def_id` methdo, you can get the `DefId` by destructuring the field of the `Trait` and `Mod` variants.